### PR TITLE
App Crash during Scroll Animation while navigating away from Page - fix

### LINKED ISSUE #22443: App Crash on Android due to ScrollViewHandler.DisconnectHandler call during Scroll Animation while navigating away from Page.
### Description

The android application experiences a crash when a user attempts to navigate back (i.e. removing the page from the stack) while a scroll animation is in progress and the ScrollViewHandler has been disconnected.

Here are three suggested improvements:

1. Refine the RequestScrollTo CommandMapping by adding a null-check for the VirtualView within the OnFinish Action Callback. This will prevent potential null reference errors.

2. Prior to disposing of the ScrollView from the ScrollViewHandler.DisconnectHandler, cancel the _scrollCompletionSource. This will ensure that no tasks are left open, improving resource management.

3. Create a publicly (or protected) accessible method to cancel the ScrollView Scroll animation task. This will provide greater flexibility and control over the Scroll animation.

### Steps to Reproduce

1. Clone the reproduction project repository
2. Run on Android device
3. Navigate to Second Page
4. On Appearing of Second Page navigate back to main page (before scroll animation finishes)
5. App will crash

### Link to public reproduction project repository

https://github.com/PATRICKdallat/ScrollViewHandlerCrash

### Version with bug

8.0.10 SR3

### Is this a regression from previous behavior?

No, this is something new

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

_No response_

### Did you find any workaround?

Replacing the CommandMapper for IScrollView.RequestScrollTo

`        ScrollViewHandler.CommandMapper.ReplaceMapping<IScrollView, ScrollViewHandler>(nameof(IScrollView.RequestScrollTo), (handler, _, args) => 
        {
            var context = handler.PlatformView.Context;
            
            if (args is not ScrollToRequest request || context is null)
            {
                return;
            }
        
            var horizontalOffsetDevice = (int)context.ToPixels(request.HorizontalOffset);
            var verticalOffsetDevice = (int)context.ToPixels(request.VerticalOffset);
        
            handler.PlatformView.ScrollTo(horizontalOffsetDevice, verticalOffsetDevice,
                request.Instant, () => handler?.VirtualView?.ScrollFinished());
        });`

### Relevant log output

`Android.Runtime.RuntimeNativeMethods.monodroid_debugger_unhandled_exception (Unknown Source:0)
Android.Runtime.JNINativeWrapper._unhandled_exception (JNINativeWrapper.g.cs:13)
Android.Runtime.JNINativeWrapper.Wrap_JniMarshal_PPL_V (JNINativeWrapper.g.cs:126)
Microsoft.Maui.Handlers.ViewHandler<Microsoft.Maui.IScrollView,Microsoft.Maui.Platform.MauiScrollView>.get_VirtualView (ViewHandlerOfT.cs:42)
Microsoft.Maui.Handlers.ScrollViewHandler.Microsoft.Maui.Handlers.IScrollViewHandler.get_VirtualView (ScrollViewHandler.cs:53)
Microsoft.Maui.Handlers.ScrollViewHandler. (ScrollViewHandler.Android.cs:150)
Microsoft.Maui.Platform.MauiScrollView. (MauiScrollView.cs:288)
Android.Animation.AnimatorEventDispatcher.OnAnimationEnd (Animator.cs:86)
Android.Animation.Animator.IAnimatorListenerInvoker.n_OnAnimationEnd_Landroid_animation_Animator_ (Android.Animation.Animator.cs:142)
Android.Runtime.JNINativeWrapper.Wrap_JniMarshal_PPL_V (JNINativeWrapper.g.cs:125)`

### PR DESCRIPTION
**Pull request by kubaflo**
### Issues Fixed

Fixes https://github.com/dotnet/maui/issues/22443

### CODE REVIEW

File: src/Core/src/Handlers/ScrollView/ScrollViewHandler.Android.cs [hunk 1]
```diff
@@ -147,7 +147,7 @@ public static void MapRequestScrollTo(IScrollViewHandler handler, IScrollView sc
 			var verticalOffsetDevice = (int)context.ToPixels(request.VerticalOffset);
 
 			handler.PlatformView.ScrollTo(horizontalOffsetDevice, verticalOffsetDevice,
-				request.Instant, () => handler.VirtualView.ScrollFinished());
+				request.Instant, () => scrollView?.ScrollFinished());
```
Comment (PureWeen) [inline src/Core/src/Handlers/ScrollView/ScrollViewHandler.Android.cs:150]: can we leave the previous code but do a check to see if the handler is still connected?

```C#
if (this.Handler.IsConnected())
      handler.VirtualView.ScrollFinished()
```

**Review by PureWeen — APPROVED** [review]

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue22443.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+public class Issue22443 : _IssuesUITest
+{
+    public Issue22443(TestDevice device)
+		: base(device)
+	{ }
+
+    public override string Issue => "App Crash on Scroll Animation while navigating away from Page";
+
+    [Test]
+	public void NoExceptionShouldBeThrown()
```
Comment (PureWeen) [inline src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue22443.cs:16]: Test is failing on catalyst

**Review by PureWeen — CHANGES_REQUESTED** [review]

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue22443.cs (hunk 2 above)
Comment (kubaflo) [reply src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue22443.cs:16]: Screenshot 2024-05-19 at 00 57 28
Should work now :)

**Review by PureWeen — APPROVED** [review]

**Review by jsuarezruiz — APPROVED** [review]

**Comment by PATRICKdallat** [comment]
Not a fix, just a cover up/workaround.
Please take a look at my comment in the bug ticket linked below.
https://github.com/dotnet/maui/issues/22443#issuecomment-2120605836

**Comment by PureWeen** [comment]
Failing test isn't related

**Pull request merged**